octopus-agent/src/main/java/octopus/teamcity/agent/buildinformation/OctopusBuildInformationBuildProcess.java

```diff
@@ -52,9 +52,8 @@ public OctopusBuildInformationBuildProcess(
     final Map<String, String> sharedConfigParameters = runningBuild.getSharedConfigParameters();
 
     final BuildInfoUserData buildInfoUserData = new BuildInfoUserData(parameters);
-    final String buildId = Long.toString(runningBuild.getBuildId());
 
-    final URL buildUrl = constructBuildUrl(runningBuild, buildId);
+    final URL buildUrl = constructBuildUrl(sharedConfigParameters.get("externalBuildURL"));
```

Comment:
I was thinking about this - we COULD just pass a String through to the payload - there's no real need to validate this is a correctly formed URL - we're just a opaque pipe, routing data from TC --> OD.
I.e. never convert this data into a URL object, it doesn't really add value now that we're not modifying it

Reply:
I don't mind it being a Url, strong types aren't a bad thing